lib/metrics/active_users.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module Metrics
+  # Utility class to generate and publish a set of metrics for the
+  # provided period and date arguments. It delegates the actual
+  # generation to PerformancePlatform::Gateway::ActiveUsers and will
+  # upload the result in the S3 bucket designated through
+  # ENV['S3_METRICS_BUCKET'].
+  class ActiveUsers
+    VALID_PERIODS = %w[week day month].freeze
+
+    attr_reader :period, :date, :report, :s3
+
+    def initialize(attrs)
+      raise ArgumentError unless VALID_PERIODS.include? attrs[:period]
+
+      @period = attrs[:period]
+      @date = attrs[:date]
+
+      @s3 = Aws::S3::Client.new(region: 'eu-west-2')
+    end
+
+    def generate!
+      gateway = PerformancePlatform::Gateway::ActiveUsers.new(
+        period: period,
+        date: date
+      )
+
+      @report = gateway.fetch_stats
+    end
+
+    def key
+      "logging-api-active-users-#{period}-#{date}"
```

Comment:
My preference is this, in support of keeping the data decoupled from the code which generates it (as the code may very well move to another place in future, and the data isn't linked to logging API):
```suggestion
      "active-users-#{period}-#{date}"
```